packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/02-implementation.md

```diff
@@ -1,17 +1,22 @@
 ---
-title: Implementation
+title: Building an Upgradeable Plugin
 ---
 
-## Implementing the Logic
+## How to build an Upgradeable Plugin implementation contract
 
-Since we took care of the initialization of our upgradeable plugin, we can now implement the logic.
+In this guide, we'll build a `SimpleStorage` Upgradeable plugin which all it does is storing a number.
```

Comment:
Later, it also stores an address as well and emits events. Not sure when it's best to mention that

Reply:
We mention it when we speak about the subsequent builds! At this stage, it's only the basic initial implementation so I'd leave it as is for simplicity.